src/compiler/binder.ts

```diff
@@ -1278,6 +1279,23 @@ namespace ts {
             activeLabels!.pop();
         }
 
+        function isDottedName(node: Expression): boolean {
+            return node.kind === SyntaxKind.Identifier || node.kind === SyntaxKind.ThisKeyword ||
+                node.kind === SyntaxKind.PropertyAccessExpression && isDottedName((<PropertyAccessExpression>node).expression);
+        }
+
+        function bindExpressionStatement(node: ExpressionStatement): void {
+            bind(node.expression);
+            // A top level call expression with a dotted function name and at least one argument
+            // is potentially an assertion and is therefore included in the control flow.
+            if (node.expression.kind === SyntaxKind.CallExpression) {

```

Comment:
Before I forget: Even if we don't support them, we'll need tests for our behavior of:
* Assertion constructors (arguably reasonable to support, just because construction is just calling with a replaced return value, minimally in the context of callable constructors a la js functions, it should be tested)
* Assertion tagged templates (a tagged template is _actually_ just a call, but given a valid template type signature, the most you could say is something like `function html(text: TemplateStringsArray, holes: any[]): asserts holes[0] is Something` which is... interesting.)
* Assertion decorators (I don't even know what someone would expect here - asserting the type of the class your wrapping?)
* Assertion JSX tags (lolwut - jsx tags are fake calls, so at best you could assert something about one of the props you're passed in?)
* Getter/setter pairs with assertion predicates